pkg/admission/admission.go

```diff
@@ -0,0 +1,112 @@
+package admission
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/util/quotapool"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// Config configures a Controller.
+type Config struct {
+	Limit uint64
+}
+
+// Default config settings:
+const (
+	defaultConcurrency = 50
+)
+
+// DefaultConfig returns a Config with preset values.
+func DefaultConfig() Config {
+	c := Config{
+		Limit: defaultConcurrency,
+	}
+
+	return c
+}
+
+// Controller determines the current admission parameters.
+type Controller struct {
+	Pool *quotapool.IntPool
+
+	mu struct {
+		syncutil.RWMutex
```

Comment:
no reason for the mutex if you're using atomics. 

Also, comment that you're using atomics to access these.

Reply:
Removed this. I thought I might want to add something else that can't be dealt with atomically, but ended up not doing anything with the mutex.